src/stream/src/common/table/state_table.rs

```diff
@@ -406,15 +406,16 @@ where
             .collect_vec();
 
         // Compute i2o mapping
+        // Note that this can be a partial mapping, since we use the i2o mapping to get
+        // any 1 of the output columns, and use that to fill the input column.
```

Comment:
By "to get any 1 of the output columns", does it mean that in `output_column_ids_to_input_idx` we already lost the duplicated "output pos"es?

Reply:
Yes, and that's fine.

Because we only need 1.

Consider the following output chunk from upstream:
| v1 | v1 |
| - | - |
| 1  | 1    |

And we have the upstream state table schema:
| v1 | v2 | v3 |

We just need v1, doesn't matter where it's the first or second col.
For v2, v3 we will just fill it with null.

Reply:
Ok that makes sense.

May I ask another one that, is the `StateTableInner::i2o_mapping` field only used in arrangement backfill? If yes, is it possible to store the mapping in arrangement backfill executor instead of in state table structure? In aggregate we also have a similar column mapping and I stored the mapping in `AggStateStorage::MaterializedInput::mapping` and it seem just work. I'm not sure with the arrangement backfill case though, could you explain a little bit for the initial motivation?

Reply:
> I'm not sure with the arrangement backfill case though, could you explain a little bit for the initial motivation?

It's used within replicated state table, hence we store it inside `state_table`.